rhel-3tier-iaas/azuredeploy.json

```diff
@@ -0,0 +1,951 @@
+﻿{
+   "$schema":"https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+   "contentVersion":"1.0.0.0",
+   "parameters":{
+      "adminUsername":{
+         "type":"string",
+         "metadata":{
+            "description":"Username for the Virtual Machines"
+         }
+      },
+      "adminPassword":{
+         "type":"securestring",
+         "minLength":12,
+         "metadata":{
+            "description":"Password for the Virtual Machines(Min Length: 12)"
+         }
+      },
+      "webTierVmCount":{
+         "type":"int",
+         "defaultValue":2,
+         "metadata":{
+            "description":"Number of Web servers to be deployed"
+         }
+      },
+      "appTierVmCount":{
+         "type":"int",
+         "defaultValue":2,
+         "metadata":{
+            "description":"Number of App servers to be deployed"
+         }
+      },
+      "databaseTierVmCount":{
+         "type":"int",
+         "defaultValue":2,
+         "metadata":{
+            "description":"Number of Database servers to be deployed"
+         }
+      }
+   },
+   "variables":{
+      "webTierVmSize":"Standard_D2_v2",
+      "appTierVmSize":"Standard_D2_v2",
+      "databaseTierVmSize":"Standard_D2_v2",
+      "jumpVmSize":"Standard_D1_v2",
+      "webTierStorageAccountName":"[concat('webstorage',uniqueString(resourceGroup().id))]",
+      "webTierStorageAccountType":"Standard_GRS",
+      "appTierStorageAccountName":"[concat('appstorage',uniqueString(resourceGroup().id))]",
+      "appTierStorageAccountType":"Standard_GRS",
+      "databaseTierStorageAccountName":"[concat('dbstorage',uniqueString(resourceGroup().id))]",
+      "databaseTierStorageAccountType":"Standard_GRS",
+      "jumpVmStorageAccountName":"[concat('jumpstorage',uniqueString(resourceGroup().id))]",
+      "jumpVmStorageAccountType":"Standard_LRS",
+      "diagStorageAccountName":"[concat('diagstorage',uniqueString(resourceGroup().id))]",
+      "vmStorageAccountContainerName":"vhds",
+      "virtualNetworkName":"RedHat3Tier-vnet",
+      "webTierSubnetName":"web-tier-subnet",
+      "appTierSubnetName":"app-tier-subnet",
+      "databaseTierSubnetName":"database-tier-subnet",
+      "jumpSubnetName":"jump-subnet",
+      "webNSGName":"web-tier-NSG",
+      "appNSGName":"app-tier-NSG",
+      "databaseNSGName":"database-tier-NSG",
+      "jumpNSGName":"jump-NSG",
+      "webLoadBalancerName":"web-lb",
+      "weblbIPAddressName":"web-lb-pip",
+      "weblbDnsLabel":"[concat('weblb',uniqueString(resourceGroup().id))]",
+      "webLoadBalancerIPID":"[resourceId('Microsoft.Network/publicIPAddresses',variables('weblbIPAddressName'))]",
+      "webFrontEndIPConfigID":"[concat(resourceId('Microsoft.Network/loadBalancers',variables('webLoadBalancerName')),'/frontendIPConfigurations/loadBalancerFrontEnd')]",

```

Comment:
just use resourceId() for the multi-part resourceId()s (here and below) - get rid of the unnecessary concat()